### PR TITLE
feat(promisifyFn): A new function promisifyFn that is more synchronized without chained calls. ^_^

### PR DESCRIPTION
**Pull request by zhenmang**
usage: promisify
```server.on('request', (req, res) => {
    promisify(nodeFn1).then(function(data) {
        doSomethingWith(data);
        promisify(nodeFn2).then(function(result) {
            doSomethingWith(result);
            promisify(nodeFn3).then(function(result) {
                doSomethingWith(result);
            }).catch(function(error) {
                somethingWrongWith(err);
            })
        }).catch(function(error) {
            somethingWrongWith(err);
        })
    }).catch(function(err) {
        somethingWrongWith(err);
    })
});```
==============
usage: promisifyFn
```server.on('request', async (req, res) => {
    const [ err,  data ] = await promisifyFn(nodeFn1);
    if (err) return;
    doSomethingWith(data);
    const [ error,  result ] = await promisifyFn(nodeFn2);
    if (error) return;
    doSomethingWith(result);
    const [ error3,  result3 ] = await promisifyFn(nodeFn3);
    if (error3) return;
    doSomethingWith(result3);
});```

### CODE REVIEW

**Review by zhenmang — COMMENTED** [review]
Scope of influence is promisify and promisifyFn.
promisify is the original, promisifyFn is a new Promise.

File: util.js [hunk 1]
```diff
@@ -609,58 +609,80 @@ function hasOwnProperty(obj, prop) {
 
 var kCustomPromisifiedSymbol = typeof Symbol !== 'undefined' ? Symbol('util.promisify.custom') : undefined;
 
-exports.promisify = function promisify(original) {
-  if (typeof original !== 'function')
+exports.promisify = getPromisify();
+
+exports.promisifyFn = getPromisify('promisifyFn');
```
Comment (ljharb) [inline util.js:614]: This package is for node's `util` module, in browsers. `require('util').promisifyFn` doesn't exist, and so this here must not exist.

**Review by ljharb — CHANGES_REQUESTED** [review]

File: util.js [hunk 2]
```diff
@@ -609,58 +609,80 @@ function hasOwnProperty(obj, prop) {
...
+          } else {
+            promiseResolve(value);
+          }
+        }
+      }
+      args.push(callback);
+      try {
+        original.apply(this, args);
+      } catch (err) {
+        isPromisifyFn ? promiseResolve([ err, null ]) : promiseReject(err);
```
Comment (zhenmang) [inline util.js:670]: look at here.
promisifyFn is added.

Comment (ljharb) [reply util.js:670]: Not in *node core* which dictates what this package may and may not support.

Comment (zhenmang) [reply util.js:670]: ok, tks for your patience.

**Comment by goto-bus-stop** [comment]
Since this is not a Node.js feature I'll close the PR. Thanks!

**Pull request closed without merging**